lib/src/base_impl.dart

```diff
@@ -9,83 +9,41 @@ import 'dart:convert';
 import 'exceptions.dart';
 import 'reddit.dart';
 
-String fullnameSync(RedditBase base) {
+String fullnameSync(RedditBaseInitializedMixin base) {
   if (base.data != null) {
     return base.data['name'];
   }
   return null;
 }
 
-void setData(RedditBase base, Map data) {
+void setData(RedditBaseInitializedMixin base, Map data) {
   base._data = data;
 }
 
-/// A base class for most DRAW objects which handles lazy-initialization of
-/// objects and Reddit API request state.
-abstract class RedditBase {
-  /// The current [Reddit] instance.
-  final Reddit reddit;
-
-  final RegExp _snakecaseRegExp = new RegExp("[A-Z]");
+abstract class RedditBaseInitializedMixin {
+  Reddit get reddit;
+  String get infoPath;
 
   /// Returns the raw properties dictionary for this object.
   ///
   /// This getter returns null if the object is lazily initialized.
```

Comment:
Isn't it returning a `new Map()` object?

Reply:
Actually, I'm not sure why I'm allocating a map there since one should be set when we create an instance of `RedditBaseInitializedMixin`... I'll check this.